packages/wallet/src/SingleAddressWallet.ts

```diff
@@ -213,23 +213,25 @@ export class SingleAddressWallet implements ObservableWallet {
       syncStatus: this.syncStatus
     });
     const tipBlockHeight$ = distinctBlock(this.tip$);
-    this.timeSettings$ = new PersistentDocumentTrackerSubject(
-      // TODO: it only really needs to be fetched once per epoch,
-      // so we should replace the trigger from tipBlockHeight$ to epoch$.
-      // This is a little complicated since there is a circular dependency.
-      // Some logic is needed to initiate a fetch if epoch is not available in store already.
-      coldObservableProvider({
-        cancel$,
-        equals: deepEquals,
-        provider: this.networkInfoProvider.timeSettings,
-        retryBackoffConfig,
-        trigger$: tipBlockHeight$
-      }),
-      stores.timeSettings
-    );
-    this.currentEpoch$ = currentEpochTracker(this.tip$, this.timeSettings$);
+
+    // TimeSettingsTracker requires manual refetch
+    // Use timeSettings$ to get udates. It is updated on every epoch
+    const timeSettingsManualTracker = new TimeSettingsManualTracker({
+      cancel$,
+      provider: this.networkInfoProvider.timeSettings,
+      retryBackoffConfig,
+      store: stores.timeSettings
+    });
+
+    // epochAndTimeSettingsTracker uses initial time settings fetch to create the epoch info
+    // Afterwards, each epoch change also triggers refetch of time settings
+    const epochAndTimeSettingsTracker = currentEpochTracker(this.tip$, timeSettingsManualTracker);
+    this.currentEpoch$ = new TrackerSubject(epochAndTimeSettingsTracker.pipe(map(([, currentEpoch]) => currentEpoch)));
```

Comment:
> Could maybe keep the refetch logic within SingleAddressWallet?

I have done the requested changes locally and it is indeed less invasive this way, by keeping the changes local to SingleAddressWallet. Thank you @mkazlauskas !
One problem I am facing now is that some tests are failing due to timeouts. I will investigate and push after I figure out why.

Reply:
You likely need to complete() the subject in SingleAddressWallet.shutdown

Reply:
I have done the changes.
I had to force push since the changes were not a fixup, but a different approach, so there was no value in keeping incremental changes (also i rebased on master)